aomi/seed.py

```diff
@@ -500,25 +564,96 @@ def generated(client, obj, opt):
     if resp:
         existing = resp['data']
 
-    secret_obj = {}
-    for key in obj['keys']:
-        key_name = key['name']
-        if key_name in existing and not key.get('overwrite'):
-            log("Not overwriting %s/%s" % (vault_path, key_name), opt)
-            continue
-
-        if key['method'] == 'uuid':
-            log("Setting %s to a uuid" % key_name, opt)
-            secret_obj[key_name] = str(uuid4())
-        elif key['method'] == 'words':
-            log("Setting %s to random words" % key_name, opt)
-            secret_obj[key_name] = random_word()
-        else:
-            problems("Unexpected generated secret method %s" % key['method'])
-
-    genseclen = len(secret_obj.keys())
-    if genseclen > 0:
-        update_msg = "Writing %s generated secrets to %s" % \
-                     (genseclen, vault_path)
-        log(update_msg, opt)
-        client.write(vault_path, **secret_obj)
+    if obj.get('state', 'present') == 'present':
+        secret_obj = {}
+        for key in obj['keys']:
+            key_name = key['name']
+            if key_name in existing and not key.get('overwrite'):
+                log("Not overwriting %s/%s" % (vault_path, key_name), opt)
+                continue
+
+            secret_obj[key_name] = generated_key(key, opt)
+
+        genseclen = len(secret_obj.keys())
+        if genseclen > 0:
+            update_msg = "Writing %s generated secrets to %s" % \
+                         (genseclen, vault_path)
+            log(update_msg, opt)
+            client.write(vault_path, **secret_obj)
+    else:
+        log("Removing generated secret at %s" % vault_path, opt)
+        client.delete(vault_path)
+
+
+def mount_path(client, obj, opt):
+    """Manage a Vault mountpoint"""
+    aomi.validation.mount_obj(obj)
+    path = obj['path']
+    if not validate_entry(obj, path, opt):
+        return
+
+    backends = client.list_secret_backends()
+    mounted = is_mounted(path, backends, 'generic')
+    if obj.get('state', 'present') == 'present':
+        if not mounted:
+            actually_mount(client, 'generic', path)
+            log("Mounted %s" % (path), opt)
+    else:
+        if mounted:
+            unmount(client, 'generic', path)
+            log("Mounted %s" % (path), opt)
+
+
+def duo_enable(client, backend, opt):
```

Comment:
Maybe break each of the mount types into their own namespaces. Also have a common namespace with functions like `ensure(verify_fn, action_fn, fail_fn)`